src/llscan.cc

```diff
@@ -645,11 +653,47 @@ void FindReferencesCmd::PrintReferences(SBCommandReturnObject& result,
       // Basically we need to access objects and arrays as both objects and
       // arrays.
       v8::JSObject js_obj(heap_object);
-      scanner->PrintRefs(result, js_obj, err);
+      scanner->PrintRefs(result, js_obj, err, level);
+
+      if (options->recursive_scan) {
+        if (find(already_visited_references->begin(), already_visited_references->end(), addr) != already_visited_references->end()) {
+          Settings* settings = Settings::GetSettings();
+          unsigned int padding = settings->GetTreePadding();
+          std::stringstream seen_str;
+          seen_str << std::string((level) * padding, ' ') << "|\n";
+          seen_str << std::string((level) * padding, ' ') << "+" << std::string(padding, '-') << rang::fg::red << " [seen above]" << rang::fg::reset << std::endl;
+          result.Printf(seen_str.str().c_str());
+        } else {
+          already_visited_references->push_back(addr);
+          v8::Value value(llscan_->v8(), addr);
+          ObjectScanner* scanner_ = new ReferenceScanner(llscan_, value);
```

Comment:
`ObjectScanner scanner_(llscan_, value);`, so we can drop the delete below and let C++ manage memory for us :)